src/main/scala/edu/knowitall/tool/srl/ClearSrl.scala

```diff
@@ -42,7 +42,8 @@ class ClearSrl extends Srl {
   def apply(graph: DependencyGraph): Seq[Frame] = {
     val tree = new DEPTree()
 
-    graph.nodes.zipWithIndex.foreach {
```

Comment:
This made no sense to me, so I started digging down into it.

The problem seems to be with the `zipWithIndex` call itself: It's returning a set that, despite its type as `SortedSet`, is actually unsorted (or rather, not sorted correctly). The initial iteration order is correct (which is why the index is correct), but the instance returned by `zipWithIndex` gets unsorted. This is probably a bug in either the Scala core library, or else in the Scala compiler we use (picking up the wrong / inconsistent `implicit` ordering somehow). I suspect the former.

You can also fix this by replacing this line with:
```scala
graph.nodes.iterator.zipWithIndex.foreach {
```
I'd suggest maybe doing that, and adding a TODO. And maybe filing a bug + repro case with the Scala people. :)